src/Collection.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Friendica;
+
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Psr\Log\LoggerInterface;
+
+/**
+ * The Collection classes inheriting from this abstract class are meant to represent a list of database record.
+ * The associated model class has to be provided in the child classes.
+ *
+ * Collections can be used with foreach().
+ */
+abstract class Collection implements \Iterator, \Countable, \ArrayAccess
```

Comment:
Why do we "really" need this type of class? What makes it different from a plain `$model[]` array? I don't see an advantage tbh

Reply:
Because it still is an arbitrary array. A Collection makes it clear it's a Model array. Plus it holds the `totalCount` in paginated selects.

Reply:
I think in combination with the followed explanations (type-safety), I got your point... I'd say let's have a try :-)